docs/extensions.md

```diff
@@ -152,6 +152,10 @@ class SomeBlocks {
             // Will be used as the extension's namespace.
             id: 'someBlocks',
 
+            // Core extensions only: override the default extension block colors.
+            color1: '#FF8C1A',
+            color2: '#DB6E00',
```

Comment:
Should `color3` be here?

Also, when this is called `color1`, `color2`, `color3`, to me it sounds like it's just yelling to be in the format `colors: [first, second, third]`. But that takes an additional comment making it explicit that this is an *ordered* array, not just a set of colors. So, instead, I think the names `colorPrimary`, `colorSecondary`, and `colorTertiary` would be better, along with a description of what those mean (which will be easier to remember than color1/2/3!) and maybe a link to [the color specification](https://github.com/LLK/scratch-blocks/wiki/Colors).

Also, yeah, internally I think these [*are*](https://github.com/LLK/scratch-vm/pull/2161/commits/91f0d59be078f8c120b5429352f284ce87bc8c6b#diff-38d875e1d50b3158dc96404b65275a54R785) given the names color1/color2/color3. But (assuming that can't easily be changed immediately) it's still future proofing to give more specific names to the colors in extensions. Making a change to better names later would mean either all future extension code requiring updates, or including code in the VM to support both color1/2/3 and primary/secondary/tertiary formats (which makes any future code use *either* of those formats, which is sure to be confusing).

Reply:
Yeah - I've gone back and forth with how I feel about which names to use and whether to use an array. I feel like `colorPrimary`, `colorSecondary`, and `colorTertiary` don't offer much advantage since they basically mean the same as `color1`, `color2`, and `color3` and are harder to spell.

Still, I agree that these names aren't great either. Ideally I'd like more semantically-meaningful names. I'll think on this a bit more...

Also: I agree that a link to the color spec would be a good idea. Thanks for the suggestion!

Reply:
Hmmm... semantically meaningful names... maybe we can talk to @carljbowman about this?

I think in reality the colors are as follows:
color1/colorPrimary: the base color of the block (usually the most prevalent color on the block)
color2/colorSecondary: The color of shadows/droppable menus on the block (e.g. the dark green on the boolean blocks)
color3/colorTertiary: The color of the outline of the blocks?

If these definitions are actually correct, maybe we could have something like:

baseColor, shadowColor, and outlineColor...?

![image](https://user-images.githubusercontent.com/1786240/57096539-88340f80-6ce3-11e9-84f0-ad210c0054c8.png)

Reply:
I talked to @carljbowman about this a while ago but forgot to put notes here -- sorry!

Anyway, he suggested that we stick with non-semantic names. I probably won't capture his reasoning accurately but the way I understood it is that there are a bunch of semantically-disconnected uses of each color that are hard to capture in just a few words.

Also, apparently that color spec is very out-of-date to the point that @carljbowman wasn't sure we should be linking to it since it might point people in the wrong direction.